R/db-interface.R

```diff
@@ -201,20 +195,36 @@ copy_dm_to <- function(dest, dm, ...,
     return(dm)
   }
 
-  copy_data <- build_copy_data(dm, dest, table_names_out, set_key_constraints, dest_con)
+  queries <- build_copy_queries(dest_con, dm, set_key_constraints, temporary, table_names_out)
 
-  ticker <- new_ticker("uploading data", nrow(copy_data), progress = progress)
-  new_tables <- pmap(
-    copy_data, ticker(copy_to),
-    dest = dest,
-    temporary = temporary,
-    overwrite = FALSE,
-    ...
+  # create tables
+  walk(queries$create_table_queries$sql, ~{
+    rs <- DBI::dbSendStatement(dest_con, .x)
+    DBI::dbClearResult(rs)
+  })
+
+  # populate tables
+  pwalk(
+    queries$create_table_queries[c("table", "remote_table")],
+    ~  {
+      DBI::dbAppendTable(dest_con, DBI::SQL(.y), dm[[.x]])
+    }
   )
-  names(new_tables) <- copy_data$source_name
 
+  # create indexes
+  walk(queries$index_queries$sql, ~ {
+    rs <- DBI::dbSendStatement(dest_con, .x)
+    DBI::dbClearResult(rs)
+  })
+
+  # build remote dm
+  remote_tables <-
+    queries$create_table_queries$remote_table %>%
+    set_names(queries$create_table_queries$table) %>%
+    map(tbl, src = dest_con)
```

Comment:
Which `.vars` arg are you referring to ?

Reply:
It's `vars`, a bit obscure: https://github.com/tidyverse/dbplyr/blob/63f59ace7e381fdc1ac8908cd38a40356613dff2/R/tbl-sql.R#L15

Reply:
- Use `match()` below instead of setting names here and later unnaming
- Use `map2()` with table names and column names

```suggestion
  map(..., ~ tbl(dest_con, .x, vars = .y))
```